components/tabs/tab-mixin.js

```diff
@@ -0,0 +1,219 @@
+import { classMap } from 'lit/directives/class-map.js';
+import { getFocusPseudoClass } from '../../helpers/focus.js';
+import { SkeletonMixin } from '../skeleton/skeleton-mixin.js';
+
+const keyCodes = {
+	ENTER: 13,
+	SPACE: 32
+};
+
+export const TabMixin = superclass => class extends SkeletonMixin(superclass) {
+	static get properties() {
+		return {
+			ariaSelected: { type: String, reflect: true, attribute: 'aria-selected' },
+			// eslint-disable-next-line lit/no-native-attributes
+			role: { type: String, reflect: true },
+			selected: { type: Boolean, reflect: true },
+			tabIndex: { type: Number, reflect: true },
+		};
+	}
+
+	static styles = css`
+        :host {
+            box-sizing: border-box;
+            display: inline-block;
+            max-width: 200px;
+            outline: none;
+            position: relative;
+            vertical-align: middle;
+        }
+        .d2l-tab-text {
+            margin: 0.5rem;
+            overflow: hidden;
+            padding: 0.1rem;
+            text-overflow: ellipsis;
+            white-space: nowrap;
+        }
+        :host(:first-child) .d2l-tab-text {
+        	margin-inline-start: 0;
+			margin-inline-end: 0.6rem;
+        }
+        .d2l-tab-selected-indicator {
+            border-top: 4px solid var(--d2l-color-celestine);
+            border-top-left-radius: 4px;
+            border-top-right-radius: 4px;
+            bottom: 0;
+            display: none;
+            margin: 1px 0.6rem 0 0.6rem;
+            position: absolute;
+            transition: box-shadow 0.2s;
+            width: calc(100% - 1.2rem);
+        }
+        :host(:first-child) .d2l-tab-selected-indicator {
+			margin-inline-start: 0;
+			margin-inline-end: 0.6rem;
+			width: calc(100% - 0.6rem);
+		}
+        :host(:${unsafeCSS(getFocusPseudoClass())}) > .d2l-tab-text {
+            border-radius: 0.3rem;
+            box-shadow: 0 0 0 2px var(--d2l-color-celestine);
+            color: var(--d2l-color-celestine);
+        }
+        :host([aria-selected="true"]:focus) {
+            text-decoration: none;
+        }
+        :host(:hover) {
+            color: var(--d2l-color-celestine);
+            cursor: pointer;
+        }
+        :host([aria-selected="true"]:hover) {
+            color: inherit;
+            cursor: default;
+        }
+		:host([aria-selected="true"]) .d2l-tab-selected-indicator {
+			display: block;
+		}
+    `;
+
+	constructor() {
+		super();
+		this.role = 'tab';
+		this.#selected = false;
+		this.tabIndex = -1;
+
+		this.#handleClickBound = this.#handleClick.bind(this);
+		this.#handleKeydownBound = this.#handleKeydown.bind(this);
+		this.#handleKeyupBound = this.#handleKeyup.bind(this);
+	}
+
+	get ariaSelected() {
+		return this.selected;
+	}
+
+	set ariaSelected(_) {
+		// ariaSelected is a derivative of `selected` and should not be set directly
+	}
+
+	get selected() {
+		return this.#selected;
+	}
+
+	set selected(value) {
```

Comment:
Do we need this custom setter? It seems like we could set the `aria-selected` below in `update`, and the `d2l-tab-selected` event is already dispatched there.